nic: clarify config-safety (#1857)
nginx/documentation: content/nic/install/helm/parameters.md

| **controller.hostNetwork** | Enables NGINX Ingress Controller pods to use the host's network namespace. | false |
| **controller.dnsPolicy** | DNS policy for NGINX Ingress Controller pods. | ClusterFirst |
| **controller.nginxDebug** | Enables debugging for NGINX. Uses the `nginx-debug` binary. Requires `error-log-level: debug` in the ConfigMap via `controller.config.entries`. | false |

> | **controller.enableConfigSafety** | Enhances the stability and reliability of the NGINX Ingress Controller. When enabled, it ensures the validation of new configurations before reloading nginx. If a configuration is deemed invalid, the new config for associated resource is rejected and old working config is restored. This is an experimental feature. Behavior and configuration may change in future releases. When enabled, users can experience delayed pod startup times as resources are validated prior to being written, and delays will be directly proportional to number and complexity of resources. | false |

Reviewer: ```suggestion
| **controller.enableConfigSafety** | Enhances the stability and reliability of the NGINX Ingress Controller. When this feature is turned on, it ensures the validation of new configurations before reloading nginx. If a configuration is deemed invalid, the new config for the associated resource is rejected and the old working config is restored. This is an experimental feature. Behavior and configuration may change in future releases. When turned on, users can experience delayed pod startup times as resources are validated prior to being written. Delays are directly proportional to the number and complexity of resources. | false |
```
